include/ProgramVariable.h

```diff
@@ -17,6 +17,12 @@ class ProgramVariable {
   // under-approximation of must-aliases
   std::list<ProgramVariable> aliases;
 
+  // bool indicates whether the generated ProgramVariable is actually a program
+  // variable. It can be the case that the *Value is just the number 0, which
```

Comment:
by "just the number 0", do you mean that the `Value*` is nullable? Or that it's literally a reference to a zero, either in the IR or in the program itself?

Reply:
It's literally the number zero; this code was written when I saw in the IR the instruction `store i32 0, i32* %retval, align 4`. My alias reasoning initially assumed that whatever we were storing into `%retval` would be a variable/identifier and that we should store the 0 as a ProgramVariable that's aliased to `%retval`, but it not correct to store the number zero as a variable.
I found this assumption quite troubling when in that same IR, the instruction `%0 = load i8**, i8*** %p, align 8, !dbg !23` was present. My code finds aliases between variables by looking at cleaned (no % or @ as the first character) variable names. For this instruction, we would be aliasing %0 with %p and their cleaned names would be 0 and p. But the program had mistakenly thought the %0  was referring to the ProgramVariable 0, messing up the aliasing reasoning.

Reply:
Thanks, that makes sense. I think the comment you wrote in the code here is a bit confusing, so moving at least some of this explanation into the comment in the code would be helpful.